service/worker/deployment/deployment_activities.go

```diff
@@ -27,26 +27,59 @@ package deployment
 import (
 	"context"
 
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/sdk/activity"
+	sdkclient "go.temporal.io/sdk/client"
+	deployspb "go.temporal.io/server/api/deployment/v1"
 	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/primitives"
+	"go.temporal.io/server/common/sdk"
 )
 
 type (
 	DeploymentActivities struct {
 		activityDeps
-		namespace   namespace.Name
-		namespaceID namespace.ID
+		namespaceName namespace.Name
+		namespaceID   namespace.ID
 	}
 
-	StartDeploymentNameWorkflowActivityInput struct {
-		NamespaceName  string
-		NamespaceID    string
+	DeploymentNameWorkflowActivityInput struct {
 		DeploymentName string
-		// TODO Shivam - confirm if we need a buildID to be passed or not?
 	}
 )
 
 // StartDeploymentNameWorkflow activity starts a DeploymentName workflow
-func (a *DeploymentActivities) StartDeploymentNameWorkflow(ctx context.Context, input StartDeploymentNameWorkflowActivityInput) error {
-	// TODO Shivam: Fill in the implementation to start DeploymentName Workflow
+func (a *DeploymentActivities) StartDeploymentNameWorkflow(ctx context.Context, input DeploymentNameWorkflowActivityInput) error {
+	logger := activity.GetLogger(ctx)
+	logger.Info("activity to start DeploymentName workflow started")
+
+	sdkClient := a.ClientFactory.NewClient(sdkclient.Options{
+		Namespace:     a.namespaceName.String(),
+		DataConverter: sdk.PreferProtoDataConverter,
+	})
+	// Workflow options for inputting workflowID and memo and duplication policy
+	workflowID := DeploymentWorkflowIDPrefix + DeploymentWorkflowIDDelimeter + input.DeploymentName
```

Comment:
I had thought of that while drafting this up and had decided against it because we are escaping `input.DeploymentName` in the initial Deployment Workflow. I can make the changes if you think it's good practice but I can't seem to think of a world where this input won't have an escaped sequence

Reply:
sorry, yes. Changed.